src/app/modules/main/communities/module.nim

```diff
@@ -200,6 +219,20 @@ method reorderCommunityCategories*(self: Module, communityId: string, categoryId
 method communityMuted*(self: Module, communityId: string, muted: bool) =
   self.view.model().setMuted(communityId, muted)
 
+method discordCategoriesAndChannelsExtracted*(self: Module, categories: seq[DiscordCategoryDto], channels: seq[DiscordChannelDto], oldestMessageTimestamp: int) =
+  self.view.discordCategoriesModel().clearItems()
+  self.view.discordChannelsModel().clearItems()
+  self.view.setDiscordOldestMessageTimestamp(oldestMessageTimestamp)
+
+  for discordCategory in categories:
+    self.view.discordCategoriesModel().addItem(self.getDiscordCategoryItem(discordCategory))
+  for discordChannel in channels:
+    self.view.discordChannelsModel().addItem(self.getDiscordChannelItem(discordChannel))
+  self.view.discordChannelsModel().hasSelectedItemsChanged()
+
+method extractDiscordCategoriesAndChannelsFailed*(self: Module) =
+  self.view.setExtractDiscordCategoriesAndChannelsFailed(true)
```

Comment:
These are the two signals we get when calling `requestExtractDiscordCAtegoriesAndCHannels` on status-go